game-core/src/main/java/games/strategy/util/PointFileReaderWriter.java

```diff
@@ -188,6 +188,69 @@ public static void writeOneToMany(final OutputStream sink, final Map<String, Lis
     return mapping;
   }
 
+  /**
+   * Writes the specified one-to-many mapping between names and (points, overflowToLeft) to the specified stream.
+   *
+   * @param sink The stream to which the name-to-points mappings will be written.
+   * @param mapping The name-to-points mapping to be written.
+   *
+   * @throws IOException If an I/O error occurs while writing to the stream.
+   */
+  public static void writeOneToManyPlacements(final OutputStream sink,
+      final Map<String, Tuple<List<Point>, Boolean>> mapping)
+      throws IOException {
+    checkNotNull(sink);
+    checkNotNull(mapping);
+
+    final StringBuilder out = new StringBuilder();
+    final Iterator<String> keyIter = mapping.keySet().iterator();
+    while (keyIter.hasNext()) {
+      final String name = keyIter.next();
+      out.append(name).append(" ");
+      final Collection<Point> points = mapping.get(name).getFirst();
+      final boolean overflowToLeft = mapping.get(name).getSecond();
```

Comment:
This is a smell. You should iterate over the entrySet of this map and not try to iterate over the keys and lookup the values in a different step 2 times.

Reply:
Same as above. I left it using a key iterator for consistency as that is what several other methods in this class do.